src/Forms/Form.php

```diff
@@ -1334,27 +1334,12 @@ public function validate(): ValidationResult
      *  do not want them parsed as submitted data. MERGE_AS_SUBMITTED_VALUE does the opposite and forces the data to be
      *  parsed as it would be submitted from a form.
      *
-     *  For backwards compatibility reasons, this parameter can also be set to === true, which is the same as passing
-     *  MERGE_CLEAR_MISSING
-     *
      * @param array $fieldList An optional list of fields to process.  This can be useful when you have a
      * form that has some fields that save to one object, and some that save to another.
      * @return $this
      */
-    public function loadDataFrom($data, $mergeStrategy = 0, $fieldList = null)
+    public function loadDataFrom(object|array $data, int $mergeStrategy = 0, array $fieldList = [])
     {
-        if (!is_object($data) && !is_array($data)) {
-            user_error("Form::loadDataFrom() not passed an array or an object", E_USER_WARNING);
-            return $this;
-        }
```

Comment:
Adding strong typing here seems sensible.

- `object|array` is already enforced for `$data` by the user_error.
- `int` is now the only valid type for `$mergeStrategy`
- `array` is already enforced for `$fieldList` by its use in `in_array()`